src/block/block_ext.c

```diff
@@ -384,8 +395,9 @@ __block_off_remove(
         *extp = ext;
 
     /* Update the cached end-of-list. */
-    if (el->last == ext)
-        el->last = NULL;
+    if (el->last == ext) {
+        el->last = penultimate_extp;
+    }
```

Comment:
Style: We do not put { } around single statement conditionals please.

Reply:
done, thanks.